awsx/ecs/containers.ts

```diff
@@ -101,7 +101,9 @@ export function getMappingInputs(
   tgPort: number | undefined,
 ): schema.TaskDefinitionPortMappingInputs {
   return {
-    ...mappingInput,
```

Comment:
This makes me uneasy because schema.TaskDefinitionPortMappingInputs specified this schema:

```typescript
export interface TaskDefinitionPortMappingInputs {
    readonly appProtocol?: pulumi.Input<TaskDefinitionPortMappingAppProtocolInputs>;
    readonly containerPort?: pulumi.Input<number>;
    readonly containerPortRange?: pulumi.Input<string>;
    readonly hostPort?: pulumi.Input<number>;
    readonly name?: pulumi.Input<string>;
    readonly protocol?: pulumi.Input<string>;
    readonly targetGroup?: pulumi.Input<aws.lb.TargetGroup>;
}
```

So the previous code was type-safe and assumed that a targetGroup can be passed here. If we believe this is in error, then the type is likely still erroneous? Reading the codebase further, not one I'm very familiar with yet.

Reply:
Ah.. I think I found it. 

There's a lot of object merging that ends to this line:

https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/ec2TaskDefinition.ts#L111

Looks like we pollute this with the targetGroup like you point out. 

https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/ec2TaskDefinition.ts#L84

https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/containers.ts#L82

Here we're extracting the port from the targetGroup:

https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/containers.ts#L69

And then passing to getMappingInputs.

So it's entirely safe to drop targetGroup here because https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/containers.ts#L82 builds a schema.TaskDefinitionContainerDefinitionInputs that does not need or have a targetGroup.

I think this is fine then!

I wish I knew how to take a JSON blob in TypeScript and drop out all the fields that are not specified in type T, that could be handy here.

TO our previous discussion, I think in most language other than TypeScript this provider has a serialization boundary and the serde framework accomplishes the same result, namely that schema-unlisted fields may drop out.

Reply:
I think the root cause of this is because the same types that are used as input types to the component are passed directly to the underlying resource, but those types don't actually match.

In the case of `TaskDefinitionPortMappingInputs`, this is the type that the user provides as an input to the component. We then use this same type as is and pass it to the [Port Mappings](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html) property on the container definition. This is what ends up causing the issue because the `TaskDefinitionPortMappingInputs` property doesn't exactly match the `PortMapping` property (the `targetGroup` property is the extra property). 

I think the ideal solution would be to create interfaces for the `containerDefinition` property (which is currently a JSON string) and then first convert these input properties to the containerDefinition property before stringifying it. I decided not to go down that route yet because ideally that would live in the aws classic provider.

Reply:
> TO our previous discussion, I think in most language other than TypeScript this provider has a serialization boundary and the serde framework accomplishes the same result, namely that schema-unlisted fields may drop out.

I'm not sure that would even catch this case because the schema property type is a `string` and we do pass a string, it just has extra properties.